Abstract Section (#19)
greenelab/text_mined_hetnet_manuscript: content/01.abstract.md

## Abstract {.page_break_before}

Knowledge bases support multiple research efforts including providing contextual information for biomedical entities, constructing networks, and supporting the interpretation of high-throughput analyses.

> Some knowledge bases are automatically constructed, but most are populated via some semblanace of manual curation.

Reviewer: ```suggestion
Some knowledge bases are automatically constructed, but most are populated via some form of manual curation.
```

## Abstract {.page_break_before}

> Knowledge-bases are useful resources that aide in multiple processes such as network construction and analysis, providing researchers contextual information, and more.

Reviewer: `Knowledge-bases` Check that this remains the same throughout the paper

> Knowledge-bases are useful resources that aide in multiple processes such as network construction and analysis, providing researchers contextual information, and more.

Reviewer: ```suggestion
Knowledge bases support multiple research efforts including providing contextual information for biomedical entities, constructing networks, and supporting the interpretation of high-throughput analyses.
```

> Majority of these knowledge-bases are populated through manual curation.

Reviewer: I'm not entirely sure what you're going for here. It would be `A majority` or `Most` from a grammar point of view. My uncertainty is deeper though.

Do you mean that there are knowledge bases that are entirely automatically derived, but most are manually derived? Or do you mean that of knowledge bases, they include edges - some of which are automatically derived but the majority are manually curated?

Author: From my understanding, I was trying to say that most knowledge bases that include edges are populated via some form of manual curation.

> This process is a time consuming and on the verge of being infeasible as the publication rate constantly rises.

Reviewer: ```suggestion
Manual curation is time consuming and difficult to scale in the context of an increasing publication rate.
```

> A recently discovered paradigm, called Data Programming, can circumvent this arduous process.

Reviewer: ```suggestion
A recently described "data programming" paradigm seeks to circumvent this arduous process by combining distant supervision with simple rules and heuristics written as labeling functions that can be automatically applied to inputs.
```

> This paradigm uses simple rules and heuristics via simply pythonic functions called label functions to speed up the annotation process.

Reviewer: ```suggestion
```

> A caveat towards this paradigm is that constructing a useful label function is a non trivial task.

Reviewer: ```suggestion
Unfortunately writing useful label functions requires substantial error analysis and is a non trivial task: in early efforts to use data programming we found that producing each label function could take a few days.
```

> Empirically, a useful function can take approximately a few days to construct.

Reviewer: ```suggestion
```

> To fill out a complete knowledge-base this process can take on the verge of years to complete.

Reviewer: ```suggestion
Producing a biomedical knowledge base with multiple node and edge types could take hundreds or thousands of label functions.
```

> In this paper we aim to speed up this process by determining the feasibility of re-using label functions. 

Reviewer: ```suggestion
In this paper we sought to evaluate the extent to which label functions could be re-used across edge types. 
```

> With Hetionet v1 as our knowledge-base, we evaluate this process by predicting the following edge types: Disease-associates-Gene (DaG), Compound-treats-Disease (CtD), Compound-binds-Gene (CbG), and Gene-interacts-Gene (GiG).

Reviewer: ```suggestion
We used a subset of Hetionet v1 that centered on disease, compound, and gene nodes to evaluate this approach.
```

> We start off by using database oriented label functions as our baseline model.

Reviewer: ```suggestion
We compare a baseline distant supervision model with the same distant supervision resources added to edge-type-specific label functions, edge-type-mismatch label functions, and all label functions.
```

> Using a sampling with replacement approach we randomly sample other label functions to build off the baseline model.

Reviewer: ```suggestion
```

> We report our prediction performance in the form of area under the receiver operating curve (AUROC) and area under the precision recall curve (AUPR).

Reviewer: ```suggestion
```

> Through extensive sampling we achieved decent prediction performance and determined that label functions can be reused across edge types.

Reviewer: ```suggestion
We confirmed that adding additional edge-type-specific label functions improves performance.
We also found that adding one or a few edge-type-mismatch label functions also nearly always improves performance.
Adding a large number of edge-type-mismatch label functions produces more variable performance that depends on the edge type being predicted and the edge type that is the source of the label function.
```

> Lastly, we show that this approach can add novel edges to Hetionet v1, which allows us to quickly populate Hetionet v1 with novel publications and findings.

Reviewer: ```suggestion
Lastly, we show that this approach, even on this subgraph of Hetionet, could certain novel edges to Hetionet v1 with high confidence.
We expect that its use in practice would include additional filtering and scoring methods which would further enhance precision.
```
